### DISCUSSION TITLE
potential bug in GITHUB_TOKEN permissions: github-actions bot unable to merge pull request despite the PR being approved

### DISCUSSION THREAD

**Discussion by pantelis-karamolegkos** [community/community#37963] [Actions]
I have configured my repository so that GitHub actions are able to approve PRs

![Screenshot 2022-11-01 at 11 44 10 AM](https://user-images.githubusercontent.com/50835081/199205781-14bc7b29-0dc8-4099-be1d-c44e02c1f7cd.png)

I have branch protection rule, requiring 1 approval before merging.

![Screenshot 2022-11-01 at 12 00 41 PM](https://user-images.githubusercontent.com/50835081/199208916-93ab0ee6-fdf6-4916-a6e1-9dd24eba2510.png)

I have also restricted who can directly push to the protected branch

![Screenshot 2022-11-01 at 2 10 00 PM](https://user-images.githubusercontent.com/50835081/199229485-2c8b874b-3029-4e8f-a643-581ced24d8ab.png)

However the following step fails

```
      - name: perform the merge if applicable
        env:
          GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
        shell: bash
        run: |
            echo "Approving PR..."
            gh pr review --approve ${{ github.event.issue.number }}
            echo "Merging PR..."
            gh pr merge ${{ github.event.issue.number }} --admin --squash
```

(the PR is indeed approved but the merge fails)

```
Approving PR...
Merging PR...
Message: You're not authorized to push to this branch. Visit https://docs.github.com/articles/about-protected-branches/ for more information., Locations: [{Line:1 Column:58}]
```

How can I allow `github-actions` bot to also merge the PR?

And why is the bot unable to merge the PR since it has been approved? 

**Does the restriction of direct push to branch impact this behaviour? It is not supposed to, given that it's not a direct push**

### update 1

after removing the `--admin` flag, just in case

```
Approving PR...
Merging PR...
X Pull request #199 is not mergeable: the base branch policy prohibits the merge.
To have the pull request merged after all the requirements have been met, add the `--auto` flag.
To use administrator privileges to immediately merge the pull request, add the `--admin` flag.
```

### update 2

I have added the following permissions to the `GITHUB_ACTIONS` token, without any effect whatsoever

```
permissions:
 contents: write
 pull-requests: write
 repository-projects: write
```

**Comment by antonblr** [comment]
Hi @pantelis-karamolegkos  - Have you tried to actually use the `--auto` flag it recommends? 
`gh pr merge ${{ github.event.issue.number }} --auto --squash`

Per [docs](https://cli.github.com/manual/gh_pr_merge) it seems to be the right way to do:
>--auto
Automatically merge only after necessary requirements are met
